udf/merge_scalar_user_data.sql

```diff
@@ -0,0 +1,105 @@
+CREATE OR REPLACE FUNCTION udf.merge_scalar_user_data(
```

Comment:
I suggest a docstring and a few tests as self documentation on how to use this function.

Reply:
added! thank you

Reply:
Looks great.